ui/ducks/send/send.js

```diff
@@ -0,0 +1,1325 @@
+      state.stage = SEND_STAGES.EDIT;
+      state.gas.gasLimit = action.payload.gasLimit;
+      state.gas.gasPrice = action.payload.gasPrice;
+      state.amount.value = action.payload.amount;
+      state.gas.error = null;
+      state.amount.error = null;
+      state.recipient.address = action.payload.address;
+      state.recipient.nickname = action.payload.nickname;
+      state.draftTransaction.id = action.payload.id;
+      state.draftTransaction.txParams.from = action.payload.from;
+      slice.caseReducers.updateDraftTransaction(state);
+    },
+    /**
+     * sets the provided gasLimit in state and then recomputes the gasTotal,
+     * setting that value in state as well. Once gasTotal is computed
+     * recomputes the maximum amount if the current amount mode is 'MAX' and
+     * sending the native token. ERC20 assets max amount is unaffected by
+     * gasTotal so does not need to be recomputed. Finally, validates the gas
+     * field and send state, then updates the draft transaction.
+     */
+    updateGasLimit: (state, action) => {
+      state.gas.gasLimit = addHexPrefix(action.payload);
+      state.gas.gasTotal = addHexPrefix(
+        calcGasTotal(state.gas.gasLimit, state.gas.gasPrice),
+      );
+      if (
+        state.amount.mode === AMOUNT_MODES.MAX &&
+        state.asset.type === ASSET_TYPES.NATIVE
+      ) {
+        slice.caseReducers.updateAmountToMax(state);
+      }
+      slice.caseReducers.validateGasField(state);
+      // validate send state
+      slice.caseReducers.validateSendState(state);
+      slice.caseReducers.updateDraftTransaction(state);
+    },
+    /**
+     * sets the provided gasPrice in state and then recomputes the gasTotal,
+     * setting that value in state as well. Once gasTotal is computed
+     * recomputes the maximum amount if the current amount mode is 'MAX' and
+     * sending the native token. ERC20 assets max amount is unaffected by
+     * gasTotal so does not need to be recomputed. Finally, validates the gas
+     * field and send state, then updates the draft transaction.
+     */
+    updateGasPrice: (state, action) => {
+      state.gas.gasPrice = addHexPrefix(action.payload);
+      state.gas.gasTotal = addHexPrefix(
+        calcGasTotal(state.gas.gasLimit, state.gas.gasPrice),
+      );
+      if (
+        state.amount.mode === AMOUNT_MODES.MAX &&
+        state.asset.type === ASSET_TYPES.NATIVE
+      ) {
+        slice.caseReducers.updateAmountToMax(state);
+      }
+      slice.caseReducers.validateGasField(state);
+      // validate send state
+      slice.caseReducers.validateSendState(state);
+      slice.caseReducers.updateDraftTransaction(state);
+    },
+    /**
+     * sets the amount mode to the provided value as long as it is one of the
+     * supported modes (MAX|INPUT)
+     */
+    updateAmountMode: (state, action) => {
+      if (Object.values(AMOUNT_MODES).includes(action.payload)) {
+        state.amount.mode = action.payload;
+      }
```

Comment:
I think this function should probably trigger a fresh gas estimation, right?

Reply:
Gas estimation change is handled in the action creator. This method is not exported to be dispatched directly